### DISCUSSION TITLE
Get user information on chat level

### DISCUSSION THREAD

**Discussion by felixgabler** [flyerhq/flutter_chat_ui#204] [Ideas & Requests]
I really like that we can specify users and get/show their avatar images and names on the message.
However, the current implementation does not make sense in my context and I wonder whether others have encountered the same problem or even found a solution for it.

In our app, the user information can change quite often and we would then have to update all the old messages to change their users. It would be way easier if we could specify a map/function to resolve a user ID to the user's information in the Chat widget.

What do you think? Would a PR make sense? One could in theory allow both solutions to co-exist

**Comment by demchenkoalex** [comment]
So `author` field of type `User` in the messages actually requires only 1 field - `id`. So in your case this is what you'll provide. However, with a current implementation there will be no names or avatars rendered, since they will not be found in the `author` field. What we will need to change to allow to resolve that by id on every message?

**Reply by felixgabler** [reply]
There are different ways of handling this I can imagine:

1. Pass an optional `Map<String, User>` (or `User Function(String)`) to the Chat() which will be used to resolve it using the `author.id` in the message. Less work for users but also less stylistic flexibility.
2. Pass an optional `buildAvatar` function to the Chat() that will be passed the user of each message. This gives users more flexibility but also more work.

Which one do you prefer in the context of your general approaches to these problems? I could quickly create a PR then.

**Reply by demchenkoalex** [reply]
I don't yet understand how number 1 will work, and for number 2 I guess we will also need name builder? In our project we name those like `avatarBuilder`/`nameBuilder`. Also both of them should include the colour of the avatar background/name that we generate based on an id, so we are still consistent. So I guess it would be something similar to `Widget avatarBuilder(types.User, Color)`

**Reply by felixgabler** [reply]
I will prepare two draft PRs tomorrow so you can see which solution you prefer
